tools/ci_build/github/azure-pipelines/linux-cpu-aten-pipeline.yml

```diff
@@ -28,12 +32,40 @@ jobs:
     inputs:
       versionSpec: '16.x'
 
+  - task: Cache@2
+    displayName: Cache Docker Image Task
+    inputs:
+      key: '$(TODAY) | "$(Build.SourceBranch)" | "$(Build.SourceVersion)"'
```

Comment:
Why both "$(Build.SourceBranch)" and "$(Build.SourceVersion)" are needed?

Reply:
1. Variables are needed to keep cache updated.
2. Theoretically, hash of the files is better than ADO variable. But ADO Task inputs only accept compile time variable. 
We couldn't get the compile time hash in one job, unless there's another job to create the hash. It may complicate things. 
It's the similar issue.
https://stackoverflow.com/questions/75053979/how-to-get-yesterday-in-compile-time

Reply:
Their doc says:
"Keys are composed of a combination of string values, file paths, or file patterns, where each segment is separated by a | character." 
It seems that we can pass a file path ?

Reply:
Dir name is not supported, but we could add some files path as keys.

Reply:
NO, I tried but it didn't work.

Reply:
No, I tried, but it didn't work.